### PR TITLE
feat(native-app): Add inbox module to home screen.

### PR DESCRIPTION
**Pull request by thoreyjona**
Based on top of #15331 
## What

Add inbox module to home screen in app. Also update Empty card styles and remove 'Sjá allt' link for the modules if there is an empty card.

## Screenshots / Gifs
<div>
</div>
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added `HelloModule` and `InboxModule` components to the home screen, enhancing user engagement with greetings and inbox functionalities.
  - Introduced `InboxCard`, a new UI component for displaying individual inbox items with bookmarking capabilities.

- **Updates**
  - Updated the mobile front page image retrieval through a new GraphQL query.
  - Improved text and label translations for a more user-friendly experience.
  - Enhanced visual styling and structure of empty state cards for a cleaner look.

- **Bug Fixes**
  - Corrected conditional rendering logic in the applications module for more accurate display of buttons.

### CODE REVIEW

File: apps/native/app/src/screens/home/inbox-module.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+export const InboxModule = React.memo(
+  ({ documents, loading }: InboxModuleProps) => {
+    const theme = useTheme()
+    const intl = useIntl()
+    const { getOrganizationLogoUrl } = useOrganizationsStore()
+
+    return (
+      <SafeAreaView style={{ marginTop: 16 }}>
```
Comment (snaerseljan) [inline apps/native/app/src/screens/home/inbox-module.tsx:40]: ~nit: use theme and below as well

File: apps/native/app/src/screens/home/inbox-module.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                    <ChevronRight />
+                  </TouchableOpacity>
+                )
+              }
+            >
+              <FormattedMessage id="home.inbox" />
+            </Heading>
+          </TouchableOpacity>
+          {loading ? (
+            Array.from({ length: 3 })
```
Comment (snaerseljan) [inline apps/native/app/src/screens/home/inbox-module.tsx:68]: ~nit you could wrap this within a memo

File: apps/native/app/src/ui/lib/card/inbox-card.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  onPress,
+  subject,
+  publicationDate,
+  icon,
+  unread,
+  bookmarked,
+  senderName,
+}: InboxCardProps) {
+  const theme = useTheme()
+  const [starred, setStarred] = useState<boolean>(false)
```
Comment (snaerseljan) [inline apps/native/app/src/ui/lib/card/inbox-card.tsx:31]: ~nit: Redundant boolean type. TS will infer it

File: apps/native/app/src/ui/lib/link/link-text.tsx [hunk 4]
```diff
@@ -1,24 +1,48 @@
...
+import { fontByWeight } from '@ui/utils'
 
 const Host = styled.View`
   border-bottom-width: 1px;
   border-bottom-color: ${(props) => props.theme.color.blue400};
   margin-bottom: 1px;
 `
 
+const variants = {
+  small: { fontSize: 12, lineHeight: 16, fontWeight: '600' },
```
Comment (snaerseljan) [inline apps/native/app/src/ui/lib/link/link-text.tsx:13]: you can import theme from @ui and use values from the theme

File: apps/native/app/src/ui/lib/link/link-text.tsx [hunk 5]
```diff
@@ -1,24 +1,48 @@
...
 interface LinkTextProps {
   children: string | React.ReactNode
+  variant?: LinkVariant
+  icon?: React.ReactNode | ImageSourcePropType
 }
 
-export function LinkText({ children }: LinkTextProps) {
+export function LinkText({ children, variant = 'large' }: LinkTextProps) {
   const theme = useTheme()
+  const styles = (variants as Record<string, TextStyle>)[variant]
```
Comment (snaerseljan) [inline apps/native/app/src/ui/lib/link/link-text.tsx:27]: you can cast variants to const and skip the casting of Record, i.e.

```
const variants = {
  small: { fontSize: 12, lineHeight: 16, fontWeight: '600' },
 large: { fontSize: 16, lineHeight: 20, fontWeight: '600' },
} as const
```

File: apps/native/app/src/ui/lib/link/link-text.tsx [hunk 6]
```diff
@@ -1,24 +1,48 @@
...
+  const fontSize = styles.fontSize ?? 15
```
Comment (snaerseljan) [inline apps/native/app/src/ui/lib/link/link-text.tsx:28]: fontSize should always be defined. Where does this 15 come from

File: apps/native/app/src/ui/lib/link/link-text.tsx [hunk 7]
```diff
@@ -1,24 +1,48 @@
...
+  const variantLineHeight = (styles.lineHeight ?? 1) / (styles.fontSize ?? 1)
+  const lineHeight = fontSize * variantLineHeight ?? styles.lineHeight
+  const fontFamily = fontByWeight(styles.fontWeight || '400')
```
Comment (snaerseljan) [inline apps/native/app/src/ui/lib/link/link-text.tsx:31]: 400 use from theme?

**Review by snaerseljan — APPROVED** [review]
Awesome job :)

File: apps/native/app/src/ui/lib/link/link-text.tsx (hunk 6 above)
Comment (thoreyjona) [reply apps/native/app/src/ui/lib/link/link-text.tsx:28]: sorry, copy paste from Typography component 😄

**Pull request merged**